Sources/Moya/Task.swift

```diff
@@ -28,10 +28,10 @@ public enum Task {
     case uploadFile(URL)
 
     /// A "multipart/form-data" upload task.
-    case uploadMultipart([MultipartFormData])
+    case uploadMultipart(MultipartFormData)
```

Comment:
i'm not completely familiar, but are you sure this isn't breaking?

callsites like `.uploadMultipart([literal, array, works])` will continue to compile, but `.uploadMultipart(anArrayVariable)` wouldn't?

i might be totally wrong. correct me if i am

Reply:
You are completely right.

I missed the most obvious thing 😅

This means that the PR should target `development` branch instead then?

Reply:
Maybe this would be a better solution:

```swift
/// Represents an HTTP task.
public enum Task {
    // other cases...

    /// A "multipart/form-data" upload task.
    case uploadMultipartFormData(MultipartFormData)

    /// A "multipart/form-data" upload task.
    @available(*, deprecated, message: "use `uploadMultipartFormData(MultipartFormData)` instead")
    case uploadMultipart([MultipartFormBodyPart])

    /// A "multipart/form-data" upload task  combined with url parameters.
    case uploadCompositeMultipartFormData(MultipartFormData, urlParameters: [String: Any])

    /// A "multipart/form-data" upload task  combined with url parameters.
    @available(*, deprecated, message: "use `uploadCompositeMultipartFormData(MultipartFormData)` instead")
    case uploadCompositeMultipart([MultipartFormBodyPart], urlParameters: [String: Any])
}
```

This way it doesn't break the client's implementation.

I wanted to have the same case name with different associated values, but the Swift compiler [doesn't allow it (yet)](https://github.com/apple/swift/issues/52479)

What do you think?

Reply:
yeah i think that could work, mention me when you make the change and i'll check the diff again. it might also be useful to add the two examples as test cases, to make sure both continue to work

Reply:
@AndrewSB I've added the 2 new cases and deprecated the old ones. I've updated tests to make sure both old and new cases still pass, and the documentation and example code as well.

Reply:
@AndrewSB, @sunshinejr Can you please take a look at the latest changes when you get the chance. Thanks!

Reply:
@ljubinkovicdj93 the only thing i still see that `MultipartFormData.FormDataProvider` got renamed to `MultipartFormDataPart.FormDataProvider`. i think this is good to go if you just move that enum back to be nested under `MultipartFormData` instead of the new `MultipartFormDataPart`

Reply:
@AndrewSB [Done](https://github.com/Moya/Moya/pull/2302/commits/d4578c658ad5ebd92716777d3c53a92a7f2f21c3)